activesupport/lib/active_support/messages/rotator.rb

```diff
@@ -8,6 +8,7 @@ def initialize(*, **options)
 
         @options   = options
         @rotations = []
+        @on_rotation = options[:on_rotation]
```

Comment:
Let's do this:

```ruby
def initialize(*, on_rotation: nil, **options)
```

Don't think it makes sense for the rotations to have `on_rotation` passed in. It's the root primary verifier/encryptor that owns it either when passed in through its contructor or when `verified`/`decrypt_and_verify` is called on it.

Reply:
yup agreed 👍